docs/how-to-guides.md

```diff
@@ -0,0 +1,383 @@
+  where the `dune` file lives.
+- `(dirs foo bar)`
+  ([docs](https://dune.readthedocs.io/en/stable/dune-files.html#dirs)): This
+  stanza tells Dune to only look into `foo` and `bar` subdirectories of the
+  current folder.
+
+So for example, if your library had this configuration in its `bsconfig.json`:
+
+```json
+{
+  "sources": ["src", "lib"]
+}
+```
+
+You might translate this to a `dune` file with the following configuration:
+
+```text
+(include_subdirs unqualified)
+(dirs src lib)
+(library
+ (name lib)
+ (modes melange))
+```
+
+Alternatively, depending on the case, you could place two separate `dune` files,
+one in `src` and one in `lib`, and define one `library` on each. In that case,
+`include_subdirs` and `dirs` would not be necessary.
+
+Regarding the `"type" : "dev"` configuration in ReScript, the way Dune solves
+that is with public and private libraries. If a `library` stanza includes a
+`public_name` field, it becomes a public library, and will be installed.
+Otherwise it is private and won't be visible by consumers of the package.
+
+#### `bs-dependencies`
+
+Your library might depend on other libraries. To specify dependencies of the
+library in the `dune` file, you can use the `libraries` field of the `library`
+stanza.
+
+For example, if `bsconfig.json` had something like this:
+
+```json
+"bs-dependencies": [
+  "reason-react"
+]
+```
+
+Your `dune` file will look something like:
+
+```text
+(library
+ (name lib)
+ (libraries reason-react)
+ (modes melange))
+```
+
+Remember that you will have to add all dependencies to your library `opam`
+package as well.
+
+#### `bs-dev-dependencies`
+
+There is no such thing as `bs-dev-dependencies` in Dune. If you need to bring
+some libraries or packages for development, you can just `opam install` them, or
+use the [upcoming `--with-dev-setup`
+configuration](https://opam.ocaml.org/doc/Manual.html#pkgvar-with-dev-setup) in
+opam 2.2. For example, to add `melange-jest` as dev dependency, you could add
+this in your library `opam` file:
+
+```text
+depends: [
+  "melange-jest" {with-dev-setup}
+]
+```
+
+Then, the library `melange-jest` would become available to your library tests.
+One just needs to make sure these tests are not part of a public library, if
+they are supposed to stay private.
+
+#### `pinned-dependencies`
+
+Dune allows to work inside monorepos naturally, so there is no need for pinned
+dependencies.
+[Packages](https://dune.readthedocs.io/en/stable/reference/packages.html) can be
+defined in the `dune-project` file using the `packages`, and multiple
+`dune-project` files can be added across a single codebase to work in a monorepo
+setup.
+
+#### `external-stdlib`
+
+There is no direct mapping of this functionality in Melange. If you are
+interested in it, or have a use case for it, please share with us on [issue
+melange-re/melange#620](https://github.com/melange-re/melange/issues/620).
+
+#### `js-post-build`
```

Comment:
maybe add an example of a rule that does this?

```clj
(rule 
  (deps (alias melange))
 ...
)
```

Reply:
https://github.com/melange-re/melange-re.github.io/pull/111/commits/c1f2057e2b9acfef1d35a59fc8a1487d54383c8a